### ISSUE TITLE
Proposal: "Before you interact" modal

### ISSUE THREAD

**Issue by trwnh** [tootsuite/mastodon#10384]
Context: https://lamp.institute/@dankwraith/101807672245086401

Recently an idea came up where a "before you reply" modal can be shown before replying to non-mutual posts. The idea is that this prompt would remind users to follow certain etiquette in certain situations. This can be compared to the rules/guidelines popup that appears when joining a Twitch stream, for example.

The prompt could be kept generic, but allow instances to set their own custom reminder text.

What might be more interesting is to allow users to set their own interaction prompts, which would allow users to convey their boundaries and guidelines in a more immediately communicable way than pinning a post or putting some string in their bio or profile fields.

This might even be generalized to interactions with any Object, maybe? e.g. before joining a Group, the application SHOULD display the `rules` field, and likewise for setting `inReplyTo`, etc.

The main limitations would be in determining the appropriate scope for such a feature, and also in determining the syntax for federating this information if it is customizable for Objects. Off the top of my head, the only AS property that might work is perhaps `context`? `context` is "left intentionally vague" and could therefore possibly be used for this: https://www.w3.org/TR/activitystreams-vocabulary/#dfn-context

**Comment by witcheslive** [comment]
I really like this, if anyone isn't familiar with Twitch, it does something similar when you chat in a channel the first time.

Having an interaction policy as an attribute of a profile is a great idea, actually, as many people are using limited bio space or pinned toots to do so but are not easy to access. Having dedicated profile space to an interaction policy, which can then be implemented in mastodon clients to display before first interactions and such, would be an incredibly good feature that would do quite a lot to further Mastodon as a social network that is about setting and respecting boundaries.

**Comment by ThibG** [comment]
I like the idea, but I'm worried that it would get annoying real quick. Additionally, if the message can be set by the user being replied to, it's another avenue for abuse that should be carefully considered (it's not a *big* issue, but we should at least make sure that thing is visible in the admin view).

**Comment by trwnh** [comment]
I don't think it would get annoying if it's only shown once per new person. It could also be considered unnecessary if you're already mutuals before the feature goes live?

It's fine to show this in the admin view too, as a form of user input it's moderable in the same way that any other user input is (avatar/banner/statuses/etc).

**Comment by witcheslive** [comment]
If it gets annoying that's probably a sign you're being weird, like you shouldn't be interacting with so many people you're not mutuals with that it becomes annoying. I'd say that's actually a pretty compelling reason to add it, actually.

**Comment by trwnh** [comment]
> If it gets annoying that's probably a sign you're being weird

I don't think that's the right framing. You'd only see interaction rules if that person defines them (unless the generic prompt is implemented), and in any case, it shouldn't feel weird to be made aware of someone else's boundaries.

The only case I can think of where someone might see a lot of these prompts, to the point of being excessive, is if they're a completely new user and therefore by definition making their first interactions with everyone. In that case, I think that still only affects the generic prompt and not the user-defined one. Maybe the generic prompt can have a cooldown.

**Comment by ThibG** [comment]
> If it gets annoying that's probably a sign you're being weird, like you shouldn't be interacting with so many people you're not mutuals with nor have ever talked with before in such a short span of time that it becomes annoying. I'd say that's actually a pretty compelling reason to add it, actually.

Eh, I relatively frequently receive messages from people I'm not following (some of them are following me, though, maybe that's a better criterion than being mutuals with them, for locked accounts) and I generally do not mind. I'm not sure how much I write to people with whom I'm not mutual.

I was tempted to say “I'll just implement a generic warning and see how annoying that is for me”, but turns out it's slightly trickier than expected: the client does not know whether you're following someone or not unless it especially requests that information. That could probably be easily changed for your home timeline (maybe at the cost of a few more SQL queries), but it's something else entirely for the local and public timelines.

Furthermore, this would be a client feature, and it's pretty easy imagining some of the annoying types just using a client that does not implement that feature.

**Comment by trwnh** [comment]
sure it's a client-side feature but that's kind of the whole point. a
generic warning could be added to any frontend, but a user-defined `rules`
needs api/federation modification.
-- 


*---Abdullah Tarawneh*http://trwnh.com

**Comment by trwnh** [comment]
sure it's a client-side feature but that's kind of the whole point. a generic warning could be added to any frontend, but a user-defined `rules` needs api/federation modification.

**Comment by nightpool** [comment]
I'm tentatively in favor of a server-admin setting for this

**Comment by trwnh** [comment]
still interested in this

**Comment by ryliejamesthomas** [comment]
I think this would be a great way to make the interaction guidelines some people put in their bios and/or pinned posts easier to see.

**Comment by naught101** [comment]
I think this would be better as a small banner over the reply-composition text box. Modals are super annoying. It is a relevant and useful thing to be informed about, but interacting with strangers is often a lovely part of mastodon, and a modal might reduce the frequency of that.
